diff --git a/es-app/src/guis/GuiSystemInformation.cpp b/es-app/src/guis/GuiSystemInformation.cpp
index 7830481b..00ebb130 100644
--- a/es-app/src/guis/GuiSystemInformation.cpp
+++ b/es-app/src/guis/GuiSystemInformation.cpp
@@ -701,7 +701,7 @@ void GuiSystemInformation::openSoftware()
 	s->addWithLabel(_("SO BASE"), std::make_shared<TextComponent>(window, si.so_base, font, color));
 
 	// kernel
-	s->addWithLabel(_("KERNEL"), std::make_shared<TextComponent>(window, si.linux, font, color));
+	s->addWithLabel(_("KERNEL"), std::make_shared<TextComponent>(window, si.Linux, font, color));
 
 	window->pushGui(s);
 }
diff --git a/es-core/src/platform.cpp b/es-core/src/platform.cpp
index 696a3a1d..60e2f5f8 100644
--- a/es-core/src/platform.cpp
+++ b/es-core/src/platform.cpp
@@ -913,7 +913,7 @@ SoftwareInformation querySoftwareInformation(bool summary)
 	if (!summary)
 	{
 		si.so_base = getShOutput("es-system_inf get_base_os_info");
-		si.linux = getShOutput("es-system_inf get_kernel_info");
+		si.Linux = getShOutput("es-system_inf get_kernel_info");
 	}
 	return si;
 }
diff --git a/es-core/src/platform.h b/es-core/src/platform.h
index aa1fc8e5..87fa54a3 100644
--- a/es-core/src/platform.h
+++ b/es-core/src/platform.h
@@ -232,14 +232,14 @@ struct SoftwareInformation
 		application_name = "N/A";
 		version = "N/A";
 		so_base = "N/A";
-		linux = "N/A";
+		Linux = "N/A";
 	}
 
 	std::string hostname;
 	std::string application_name;
 	std::string version;
 	std::string so_base;
-	std::string linux;
+	std::string Linux;
 };
 
 std::string queryHostname();
