diff --git a/src/go2/display.cpp b/src/go2/display.cpp
index 32b65a0..f69833c 100755
--- a/src/go2/display.cpp
+++ b/src/go2/display.cpp
@@ -839,7 +839,8 @@ void go2_surface_blit(go2_surface_t *srcSurface, int srcX, int srcY, int srcWidt
     switch (rotation)
     {
     case GO2_ROTATION_DEGREES_0:
-        src.rotation = 0;
+        src.rotation = HAL_TRANSFORM_ROT_270;
+        //src.rotation = 0;
         break;
 
     case GO2_ROTATION_DEGREES_90:
diff --git a/src/video.cpp b/src/video.cpp
index 8eab739..df3aaf4 100755
--- a/src/video.cpp
+++ b/src/video.cpp
@@ -137,8 +137,8 @@ void video_configure(struct retro_game_geometry *geom)
         geom->max_height = 544;
         geom->max_width = 960;*/
         display = rr_display_create();
-        display_width = rr_display_width_get(display);
-        display_height = rr_display_height_get(display);
+        display_width = rr_display_height_get(display);
+        display_height = rr_display_width_get(display);
     }else {
         display = rr_display_create();
         display_width = rr_display_height_get(display);
